sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1519,9 +1519,16 @@ pi_device _pi_platform::getDeviceFromNativeHandle(ze_device_handle_t ZeDevice) {
     return nullptr;
   }
 
+  // TODO: our sub-sub-device representation is currently [Level-Zero device
+  // handle + Level-Zero compute group/engine index], so there is now no 1:1
+  // mapping from L0 device handle to PI device assumed in this function. Until
+  // Level-Zero adds unique ze_device_handle_t for sub-sub-devices, here we
+  // filter out PI sub-sub-devices.
   auto it = std::find_if(PiDevicesCache.begin(), PiDevicesCache.end(),
                          [&](std::unique_ptr<_pi_device> &D) {
-                           return D.get()->ZeDevice == ZeDevice;
+                           return D.get()->ZeDevice == ZeDevice &&
+                                  (D.get()->RootDevice == nullptr ||
+                                   D.get()->RootDevice->RootDevice == nullptr);
```

Comment:
It looks to me that you are filtering out both sub-devices and sub-sub-devices.
This means you are going to return only root devices for the given ZeDevice handle.
Is that what you wanted?

Reply:
This code *allows* root-devices and sub-devices, thus filtering out sub-sub-devices *only*.

Reply:
Thanks.